kratos/includes/linear_master_slave_constraint.h

```diff
@@ -387,6 +387,28 @@ class LinearMasterSlaveConstraint
         return mMasterDofsVector;
     }
 
+    /**
+     * @brief This methgod directly applies the master/slave relationship
+     * @param rCurrentProcessInfo the current process info instance
+     */
+    void Apply(const ProcessInfo& rCurrentProcessInfo) override
```

Comment:
is rCurrentProcessInfo needed ?

Reply:
maybe not in this specific constraint, but it might be needed in the future (and changing the interface is incredibly painful => #2993 )
Same is done for elements and conditions

Reply:
+1 for having rCurrentProcessInfo.

I also would say that i quite like the "Apply"  name.

Reply:
ok

Reply:
I have nothing aginst Apply .. I just say that Apply does not indicate that it is acting on the Dofs !

If you thing that is fine ... I have no objections !

Reply:
I would suggest to avoid the allocation of the slave_dofs_values vector.

this could be done by doing

~~~cpp
Vector master_dofs_values(mMasterDofsVector.size());

        for (IndexType i = 0; i < mMasterDofsVector.size(); ++i) {
            master_dofs_values[i] = mMasterDofsVector[i]->GetSolutionStepValue();
        }

        for (IndexType i = 0; i < mSlaveDofsVector.size(); ++i) {
            double aux = mConstantVector[i];
            for(IndexType j=0; j<mRelationMatrix.size2(); ++j)
                   aux += mRelationmatrix(i,j)*slave_dofs_values[j];
                 
            #pragma omp atomic
            mSlaveDofsVector[i]->GetSolutionStepValue() += aux;
        }
~~~